cli/run_util.go

```diff
@@ -1094,20 +1094,20 @@ func runWorkflowWithConfiguration(
 	return buildRunResults, nil
 }
 
-func initializeStepDir(additionalEnvironments *[]envmanModels.EnvironmentItemModel) (string, error) {
+func createTestResultDir(additionalEnvironments *[]envmanModels.EnvironmentItemModel) (string, error) {
 	// ensure a new testDirPath and if created successfuly then attach it to the step process by and env
-	testDirPath, err := ioutil.TempDir(os.Getenv("BITRISE_TEST_RESULTS_DIR"), "test_result")
+	testDirPath, err := ioutil.TempDir(os.Getenv(configs.BitriseTestResultDirEnvKey), "test_result")
 	if err != nil {
 		return "", err
 	}
 	// managed to create the test dir, set the env for it for the next step run
 	*additionalEnvironments = append(*additionalEnvironments, envmanModels.EnvironmentItemModel{
```

Comment:
you return the `testDirPath`, append it to `additionalEnvironments` outside, this way you do not modify input parameters